admin-management-service/src/main/java/com/scaler/adminmanagementservice/controllers/AdminController.java

```diff
@@ -0,0 +1,54 @@
+package com.scaler.adminmanagementservice.controllers;
+
+import com.scaler.adminmanagementservice.dtos.GenericAdminDto;
+import com.scaler.adminmanagementservice.services.AdminService;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/admins")
+public class AdminController implements AdminOperations{
+
+    private AdminService adminService;
+
+    public AdminController(AdminService adminService){
+
+        this.adminService = adminService;
+    }
+
+
+
+    @GetMapping("{id}")
+    public GenericAdminDto getAdminById(@PathVariable("id") Long id) {
+
+        return adminService.getAdminById(id);
+    }
+
+
+    @GetMapping
+    public List<GenericAdminDto> getAllAdmins(){
+
+        return adminService.getAllAdmins();
+    }
+
+
+    @PostMapping
+    public GenericAdminDto createAdmin(@RequestBody GenericAdminDto admin) {
+
+        return adminService.createAdmin(admin);
+    }
+
+
+    @DeleteMapping("{id}")
+    public GenericAdminDto deleteAdmin(@PathVariable("id") Long id){
```

Comment:
Please take a look at this again thoroughly - https://www.baeldung.com/spring-interface-driven-controllers#2-interface

where GetMapping is used - interface or class
where override is used - interface or class
where PathVariable is used - interface of class